core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -533,11 +577,32 @@ protected <T extends RESTResponse> T execute(
     throw new RESTException("Unhandled error: %s", error);
   }
 
+  /**
+   * Supplied interface to allow RESTCatalogAdapter implementations to have a mechanism to change
+   * how many file scan tasks get grouped in a plan task or under what conditions a table scan
+   * should be performed async. Primarily used in testing to allow overriding more deterministic
+   * ways of planning behavior.
+   */
+  public interface PlanningBehavior {
+    default int numberFileScanTasksPerPlanTask() {
+      return 100;
+    }
+
+    default boolean shouldPlanTableScanAsync(TableScan tableScan) {
+      return false;
+    }
+  }
+
+  protected PlanningBehavior planningBehavior() {
+    return new PlanningBehavior() {};
+  }
```

Comment:
can we have a setter for PlanningBehaviour too incase we wanna override this behaviour ?

Reply:
Do we need a setter? https://github.com/singhpk234/iceberg/pull/271/files#diff-37f64b622d5c3e0e890d94773e84ef49ceb3c1e9b0c03da313db9266b0e75afbR187 overriding should just work via inheritance

Reply:
Yeah it should just be a one time override. If you take a look at that PR to your branch, we should be able to exercise all parts of the protocol (in your existing tests) through this. If it ends up being more complicated, we can end up adding a setter but for now I'd probably keep it as is until we know we need it.